generate/codegen/modulewriter.go

```diff
@@ -16,19 +16,29 @@ import (
 
 // ModuleWriter mantains module level auto-generated code source files
 type ModuleWriter struct {
-	Module      modules.Module
-	Description string
-	DocURL      string
-	PlatformDir string
-	Protocols   []*typing.ProtocolType
-	EnumAliases []*AliasInfo
-	FuncAliases []*AliasInfo
+	Module        modules.Module
+	Description   string
+	DocURL        string
+	PlatformDir   string
+	Protocols     []*typing.ProtocolType
+	EnumAliases   []*AliasInfo
+	FuncAliases   []*AliasInfo
+	StructAliases []*AliasInfo
+	Functions     []*Function
+	Structs       []*Struct
 }
 
 func (m *ModuleWriter) WriteCode() {
 	m.WriteDocFile()
 	m.WriteEnumAliases()
 	m.WriteTypeAliases()
+	if m.Module.Flags.GenerateStructs {
+		m.WriteStructs()
+	}
+	if m.Module.Flags.GenerateFunctions {
```

Comment:
whats the reason for putting it behind a flag?

Reply:
The rationale here is that we can add capabilities here incrementally by flagging in/on particular packages. When I was hacking on this previously I kept finding esoteric corners that were adding work.

Basic idea is that we start opt-in, get a few key packages going, and once we feel like it's ready to turn on everywhere I'd just remove the flag and press it all out.